### PR TITLE
Created routes and middleware for loading user profiles while logged in

### PR DESCRIPTION
**Pull request by mdt1026**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,63 +1,69 @@
...
 
 app.use("/user", async (req, res, next) => {
-  if(req.method === "GET") {
-      if(req.session && req.session.user) {
-          return next();
-      } else {
-          return res.redirect("/login");
+  if (req.method === "GET") {
+    if (req.session && req.session.user) {
+      console.log(req.path);
```
Comment (tbechtel40) [inline app.js]: console.log included

File: routes/user.js [hunk 2]
```diff
@@ -68,13 +76,15 @@ router
     const {username, password} = req.body;
     try{
         const user = await dbData.users.loginUser(username, password);
-        res.render('userhome', {user: user});
+        req.session.user = user;
+        res.redirect('/user/'+encodeURIComponent(user.username));
```
Comment (tbechtel40) [inline routes/user.js]: remove encodeURIComponent

**Review by tbechtel40 — CHANGES_REQUESTED** [review]

File: app.js (hunk 1 above)
Comment (mdt1026) [reply app.js]: Fixed

File: routes/user.js (hunk 2 above)
Comment (mdt1026) [reply routes/user.js]: Fixed

**Review by tbechtel40 — APPROVED** [review]

**Review by tbechtel40 — APPROVED** [review]
LGTM

**Pull request merged**